src/pieces.hpp

```diff
@@ -17,7 +17,7 @@ class Piece {
     const enum Side                 side;
 
                                     Piece(Side, int, int);
-                                    ~Piece();
+    virtual                         ~Piece();
```

Comment:
It's super important that any class used as a base class has a `virtual` descructor declared and implemented. This sets up the vtable, and ensures that objects of derived classes are correctly destroyed, even when delete is called on a point of the base class type.

More here: https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

This omission was leading to the following error in the Linux world:

```
ben@lenovo: (bedser) /scratch/schach$ bazel build //src:schach
INFO: Analysed target //src:schach (9 packages loaded).
INFO: Found 1 target...
ERROR: /scratch/schach/src/BUILD:1:1: Linking of rule '//src:schach' failed (Exit 1)
bazel-out/local-fastbuild/bin/src/_objs/schach/src/pieces.pic.o:pieces.cpp:function JumpingPiece::JumpingPiece(Side, int, int, std::vector<std::pair<int, int>, std::allocator<std::pair<int, int> > >): error: undefined reference to 'Piece::~Piece()'
bazel-out/local-fastbuild/bin/src/_objs/schach/src/pieces.pic.o:pieces.cpp:function SlidingPiece::SlidingPiece(Side, int, int, std::vector<std::pair<int, int>, std::allocator<std::pair<int, int> > >): error: undefined reference to 'Piece::~Piece()'
collect2: error: ld returned 1 exit status
Target //src:schach failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2.894s, Critical Path: 1.21s
FAILED: Build did NOT complete successfully
```

Reply:
Thanks! 
I think i will switch to gcc (if i can figure out how to do that on mac w/ bazel).